pkg/rp/util/registry_test.go

```diff
@@ -35,3 +37,69 @@ func Test_PathParserErr(t *testing.T) {
 	require.Equal(t, "", repository)
 	require.Equal(t, "", tag)
 }
+
+func Test_GetRegistrySecrets(t *testing.T) {
+	testset := []struct {
+		definition   recipes.Configuration
+		templatePath string
+		secrets      map[string]recipes.SecretData
+		exp          recipes.SecretData
+	}{
+		{
+			definition: recipes.Configuration{
+				RecipeConfig: datamodel.RecipeConfigProperties{
+					Bicep: datamodel.BicepConfigProperties{
+						Authentication: map[string]datamodel.RegistrySecretConfig{
+							"test.azurecr.io": {
+								Secret: "/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/acr",
+							},
+							"123456789012.dkr.ecr.us-west-2.amazonaws.com": {
+								Secret: "/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/ecr",
+							},
+						},
+					},
+				},
+			},
+			templatePath: "test.azurecr.io/test-private-registry:latest",
+			secrets: map[string]recipes.SecretData{
+				"/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/acr": {
+					Type: "basicAuthentication",
+					Data: map[string]string{
+						"username": "test-username",
+						"password": "test-password",
+					},
+				},
+			},
+			exp: recipes.SecretData{
+				Type: "basicAuthentication",
+				Data: map[string]string{
+					"username": "test-username",
+					"password": "test-password",
+				},
+			},
+		},
+		{
+			definition: recipes.Configuration{
+				RecipeConfig: datamodel.RecipeConfigProperties{
+					Bicep: datamodel.BicepConfigProperties{},
+				},
+			},
+			templatePath: "test.azurecr.io/test-private-registry:latest",
+			secrets: map[string]recipes.SecretData{
+				"/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/acr": {
+					Type: "basicAuthentication",
+					Data: map[string]string{
+						"username": "test-username",
+						"password": "test-password",
+					},
+				},
+			},
+			exp: recipes.SecretData{},
+		},
+	}
+	for _, tc := range testset {
```

Comment:
We need to add test verify returning error when say templatePath passed is not a valid path

Reply:
added a test